### ISSUE TITLE
No 64bit cpu detected?

### ISSUE THREAD

**Issue by richardFK** [joncampbell123/dosbox-x#3269]
### Code of Conduct & Contributing Guidelines

- [X] I agree to follow the code of conduct and the contributing guidelines.

### Have you checked that no other similar question(s) already exists?

- [X] I have searched and didn't find any similar issues.

### Question

I have started using DosBox-x (Jan 2022 x64 download on Windows 10) and just tried some 16 bit .exe from various (old) sources of mine.

I get the error "No

 64bit cpu detected" for some exe's but the other exe's (which were only assembly compiled) appear to work OK.

Is it something simple I have overlooked?

cpu is INTEL i7 x64 (Xeon)

**Comment by maron2000** [comment]
DOSBox-X (and other variants if I understand correctly) don't emulate 64-bit CPUs. 
It just recently added Pentium III emulation, which is a 32-bit CPU.

There are samples that print the same error, so you may want to check the source code.
https://www.japheth.de/JWasm/Dos64.html

**Comment by joncampbell123** [comment]
Furthermore it's extremely unlikely that any x86_64 system would have ISA slots, and the changes to the DOSBox-X CPU core would be extensive to emulate x86_64, so it is extremely unlikely DOSBox-X would ever emulate a 64-bit CPU.

There are Pentium 4 (industrial) motherboards with ISA slots though, and therefore Pentium 4 emulation may be added in the future (and SSE2).

DOSBox-X's cutoff for whether or not something is "too new" is tied to the early 2000s when the ISA bus was quickly phased out according to Microsoft "PC standards" specifications. Consumer boards dropped ISA entirely, industrial boards kept it around a bit longer.

https://flaterco.com/kb/ISA_chipsets.html

**Comment by richardFK** [comment]
@maron2000 

Thanks for info - my simple quick test included FreeBASIC windows X64 version .exe generated - to correct this I am now using FreeBASIC DOS version .exe generated instead.

So far, very simple/small FreeBASIC DOS version .exe work OK.

@joncampbell123  

Thanks for info - now understanding the situation, I have changed my development methods to suit DOSBox-X.


Please close this issue if you want.

**Comment by Torinde** [comment]
ISA slot boards - if we go very niche, I see industrial boards supporting SSE4.1 [Core 2 Quad Q9400](https://duropc.com/product/rac161) and AVX [Q77/Ivy Bridge](https://ipc.msi.com/product_detail/MS-98A9) - probably using LPC/ISA bridge chips. Even the most recent computers (or Android phones?!) can get one via PCIe/PCI/ISA bridge chips and external USB/ISA adapters [1](http://arstech.com/install/ecom-prodshow/usb2isar.html) [2](https://www.arstech.com/install/cms-display/ste_usl.html) - they even sell [dosbox-x.conf and win9x.img](https://arstech.com/install/index.php?app=ecom&ns=prodshow&ref=img-wME-w98) files! 

I wonder how do they redirect DOSbox-X to access the ISA device behind their USB bridge.

**Comment by Torinde** [comment]
No, no, I'm not asking for that, agree it would be funny though...

I'm puzzled by:
- the duropc and MSI ISA slot motherboards claiming Win7 and Win10 support (I know from experience that the Vista/2008 way to enable ISA doesn't work in Win7, so how do they make use of the ISA slots? Via some "direct access" custom drivers instead of Windows-native isapnp.sys/etc.?). WinXP support is clear, of course.
- the USB/ISA adapter - it also claims Win7/8/10 support, along with Android, etc. - but that's fine because they mention their own "software layer" and SDK (e.g. solving my first question). Here I wonder about something else - why do they sell dosbox-x.conf and win9x.img files? Obviously they use DOSbox-X in some way to access the ISA slots across their USB/ISA adapter.... but how do they achieve that only via .conf/.img files? Without any adaptation of DOSbox-X source code itself?

**Comment by Torinde** [comment]
**64-bit DOS programs**: [Necromancer's Dos Navigator](http://ndn.muxe.com/download/), [SD64 debugger](https://www.bttr-software.de/forum/forum_entry.php?id=15557) of the D3X extender, [maybe others](https://drive.google.com/drive/folders/0B_wEiYjzVkC0ZGtkbENENzF1Nms), [dosemu2/fdpp](https://github.com/dosemu2/fdpp), [Dos64-stub](https://github.com/Baron-von-Riedesel/Dos64-stub)
64-bit DOS extenders [1](https://www.bttr-software.de/forum/mix_entry.php?id=15853),[2](https://board.flatassembler.net/topic.php?t=20096),[3](http://www.masmforum.com/board/index.php?topic=8131.5), D3X, etc.
**PSE-36** DOS programs: XMS extension to access >4GB memory using [HIMEMSX](https://github.com/Baron-von-Riedesel/HimemSX), the modified ramdisk and [HDAplay](https://github.com/Baron-von-Riedesel/HDAutils)
**PSE (or PAE?)** DOS/Win9x CWSDPMI 4GB/64GB patches; PATCHMEM: Single Programs up to 2GB, but multiple Programs up to 1024GB of RAM simultaneously, RAMDISK in the extra range
**Multicore** DOS/Win9x RLoew MULTCORE; [DOS Multicore Mode Interface](https://www.codeproject.com/Articles/894522/The-Low-Level-M3ss-DOS-Multicore-Mode-Interface) https://github.com/WindowsNT/asm
**storage side**: EDR-DOS 7.01.08 support for 64-bit file systems (64 bit operations/Long SEEK), FAT+, [LBA 33-bit](https://www.bttr-software.de/forum/mix_entry.php?id=17233#p17252), multiple patches from RLoew, [gpttsd](https://github.com/LordOfMice/Tools) (GPT read/write for Win9x)

Especially relevant is the DOS Navigator version: [[1](https://www.bttr-software.de/forum/mix_entry.php?id=17233#p17252)], [actual 64-bit program](https://www.bttr-software.de/forum/mix_entry.php?id=15537), version 2 used [D3X extender](https://board.flatassembler.net/topic.php?t=14416) and didn't work with JEMM and v86. Version 3 works also with VCPI and JEMM (I'm not sure about v86).
> I think that Candyman made some update/extension to D3X extender to support 64b mode? But I don't know how/where/if it is pub. dcumented. One big disadvantege is it can only run from realmode DOS, not V86. Japhet solved this problem by updating JEMM to cooperate with himemsx. But it's not "DPMI64" just XMS64 :)

Is it possible to add support for NDN DOS64 version? @joncampbell123 explained before that PSE would be very difficult, but what about x86-64 Long mode? I think it can operate with PAE instead of PSE?

**Comment by joncampbell123** [comment]
I don't plan on emulating 64-bit long mode in DOSBox-X. If you need that, use Bochs, QEMU, or VirtualBox.

**Comment by Torinde** [comment]
I understand, but pity. I prefer DOSBox-X for various reasons...

If you allow: I see NDN is good example for a "[new DOS development](https://dosbox-x.com/newdosdevelopment.html)"
